src/utils.c

```diff
@@ -123,13 +123,15 @@ int read_derivation_path(const uint8_t *data_buffer,
     return 0;
 }
 
-void sendResponse(uint8_t tx, bool approve) {
+void sendResponse(uint8_t tx, bool approve, bool display_menu) {
     G_io_apdu_buffer[tx++] = approve ? 0x90 : 0x69;
     G_io_apdu_buffer[tx++] = approve ? 0x00 : 0x85;
     // Send back the response, do not restart the event loop
     io_exchange(CHANNEL_APDU | IO_RETURN_AFTER_TX, tx);
-    // Display back the original UX
-    ui_idle();
+    if (display_menu) {
```

Comment:
Why not use the global `G_called_from_swap` to detect if you need to update the display ? It would avoid having a new parameter to manage.

Reply:
Yes but I plan to re-use this parameter for the Stax UI that will also require a `io_exchange` without the UI call :)

Reply:
@fbeutin-ledger ah ! I thought about it after writing my comment ^^.
Ok fair enough.